packages/rs-sdk/src/sdk.rs

```diff
@@ -198,6 +207,20 @@ impl Sdk {
         self.proofs
     }
 
+    /// Set the [ContextProvider] to use.
+    ///
+    /// [ContextProvider] is used to access state information, like data contracts and quorum public keys.
+    ///
+    /// Note that this will overwrite any previous context provider.
+    pub fn set_context_provider<C: ContextProvider + 'static>(&self, context_provider: C) {
```

Comment:
Not sure I like this naming. For me context is kinda an application context, so it should be contracts for app, current user identity. Shall we call it state adapter or state repository or something like that?